openzl-crypto/src/accumulator.rs

```diff
@@ -7,7 +7,7 @@ use openzl_util::derivative;
 /// Accumulator Membership Model Types
 pub trait Types {
     /// Item Type
-    type Item: ?Sized;
+    type Item;
```

Comment:
This makes it so that we can have a concrete instance of `Item` like in the following:

```rust
struct Something<T> 
where
    T: Types,
{
    item: T::Item,
}
```

instead of having to always borrow it. It was something that was needed downstream and `?Sized` is too restrictive here and not really that helpful in most cases anyway. It can be re-introduced if we really need it.